### ISSUE TITLE
Support Gmail OAuth for mail sending

### ISSUE THREAD

**Issue by Chris8080** [mautic/mautic#1660]
Hello, 

I installed Mautic on my server for 6 different Gmail accounts (projects), sending out a couple of mails per day (10 - 30 max - in total) but running the cronjobs every 2 minutes.
Everything was working fine during the first two weeks - now all of a sudden I can't send mails via GMail anymore.
Mails are still going out via other accounts and I can send via my Gmail accounts from Thunderbird AND from a locally installed Mautic version. Just not from my server.
I assume, Gmail has blocked my IP due to too many logins?

Error log:
mautic.ERROR: Swift_TransportException: Connection could not be established with host smtp.gmail.com [ #0](uncaught exception) at vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/StreamBuffer.php line 265 while running console command `mautic:send:emails` [] []

Test connection fails with this error:
Expected response code 250 but got code "535", with message 
"535-5.7.8 Username and Password not accepted. 
Learn more at 535 5.7.8 https://support.google.com/mail/answer/14257 g132sm1136535wme.0 - gsmtp " 
Log data: ++ Starting Swift_SmtpTransport 
<< 220 smtp.gmail.com ESMTP g132sm1136535wme.0 - gsmtp >> 
EHLO s.mailr.co << 250-smtp.gmail.com at your service, 

Everything I find online is about configuration, sendmail/smtp OAuth2.0 etc. but I'm quite sure everything is completely fine (didn't change anything code wise) but Google blocked my IP due to too many logins - but I can't find anything about this online.
Any experience or suggestions?

Thanks,
Chris

**Comment by alanhartless** [comment]
Hi,

Just to verify, did you go through the steps given in the URL from the error response? https://support.google.com/mail/answer/14257

If you have two step verification enabled, you'll need to generate an app specific password. https://support.google.com/accounts/answer/185834?hl=en#ASPs

If you haven't done so already, you'll need to enable access for "less secure" applications; basically applications that simply use SMTP versus Google's oauth2 process. https://support.google.com/accounts/answer/6010255

Thanks,
Alan

**Comment by Chris8080** [comment]
Hello Alan,

yes, I went through the steps in 
https://support.google.com/mail/answer/14257

Now I have activated 2 step verification and assigned an app specific password. I'm still blocked.

"Less secure" was the first switch I used. It worked before (after using the Captcha as well).

Everything is working fine with the same setup from two other IPs - but at the server with 6 different gmail accounts I'm still blocked. Anything else that I can do?
Is it possible to "switch" Mautic into oauth2 mode? Or can I use the Gmail API instead of swiftmailer somehow?
